nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryAPI.java

```diff
@@ -4364,13 +4364,8 @@ public void testLocking() {
         assertNotNull(lock);
         assertEquals("Administrator", lock.getOwner());
 
-        // nobody can re-lock without unlock, even Administrator
-        try {
-            session.setLock(docRef);
-            fail();
-        } catch (LockException e) {
-            assertTrue(e.getMessage(), e.getMessage().contains("Document already locked"));
-        }
+        Lock relock = session.setLock(docRef);
+        assertEquals(lock, relock);
```

Comment:
Can you please also test that trying to call `setLock` on a locked document, as a user that is not the lock owner, throws the `LockException` with a 409 status code?